google/cloud/spanner_v1/database.py

```diff
@@ -525,7 +525,9 @@ def reload(self):
         self._encryption_config = response.encryption_config
         self._encryption_info = response.encryption_info
         self._default_leader = response.default_leader
-        self._database_dialect = response.database_dialect
+        # Only update if the data is specific to avoid losing specificity.
+        if response.database_dialect != DatabaseDialect.DATABASE_DIALECT_UNSPECIFIED:
```

Comment:
Will there be such a situation where backend returns UNSPECIFIED dialect?
AFAIK, the default dialect is GOOGLE_SQL. Correct me if I am missing something

Reply:
This is an issue when working with the emulator. It is returning `DATABASE_DIALECT_UNSPECIFIED`  which is overriding the dialect that this object already had. This shouldn't be an issue in production.